src/main/java/Duke.java

```diff
@@ -1,10 +1,88 @@
+            System.out.println(DIVIDING_LINE);
+            if (userInputString.equals("list")) {
+                listTasks(taskIndex, userTasks);
+            } else if (userInputString.startsWith("done")) {
+                markTaskDone(userInputString, taskIndex, userTasks);
+            } else {
+                addNewTask(userInputString, taskIndex, userTasks);
+                taskIndex++;
+            }
+            System.out.println(DIVIDING_LINE);
+            userInputString = userInput.nextLine();
+        }
+    }
+
+    public static void listTasks(int taskIndex, Task[] userTasks) {
+        for (int i = 0; i < taskIndex; i++) {
+            System.out.println((i + 1) + ". " + userTasks[i]);
+        }
+    }
+
+    public static void markTaskDone(String userInputString, int taskIndex, Task[] userTasks) {
+        String[] words = userInputString.split(" ");
+        int completeIndex = Integer.parseInt(words[1]) - 1;
+        if (completeIndex >= 0 && completeIndex < taskIndex) {
+            userTasks[completeIndex].markComplete();
+            System.out.println("Task " + userTasks[completeIndex].description + " marked as complete.");
+        } else {
+            System.out.println("Error: index outside range.");
+        }
+    }
+
+    public static void addNewTask(String userInputString, int taskIndex, Task[] userTasks) {
+        int slashIndex = userInputString.indexOf('/');
+        String taskSubstring;
+        String timeSubstring = userInputString.substring(slashIndex + 1);
+        timeSubstring = timeSubstring.replaceFirst(" ", ": ");
+        if (userInputString.startsWith("todo")) {
+            taskSubstring = userInputString.substring(TODO_OFFSET);
+            userTasks[taskIndex] = new Todo(taskSubstring);
+        } else if (userInputString.startsWith("deadline") && slashIndex > 0) {
+            taskSubstring = userInputString.substring(DEADLINE_OFFSET, slashIndex - 1);
+            userTasks[taskIndex] = new Deadline(taskSubstring, timeSubstring);
+        } else if (userInputString.startsWith("event") && slashIndex > 0) {
+            taskSubstring = userInputString.substring(EVENT_OFFSET, slashIndex - 1);
+            userTasks[taskIndex] = new Event(taskSubstring, timeSubstring);
+        } else {
+            userTasks[taskIndex] = new Todo(userInputString);
+        }
+
+        System.out.println("Gotcha. I've added this task:");
+        System.out.println(userTasks[taskIndex]);
+        System.out.println("You have a total of " + (taskIndex + 1) + " tasks now.");
+    }
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printGreeting();
+        manageTasks();
+        printFarewell();
```

Comment:
Great, easy to understand main body with correct method name representation in camelCase.